src/EditorFeatures/CSharp/CommentSelection/ToggleBlockCommentCommandHandler.cs

```diff
@@ -0,0 +1,428 @@
+
+        protected override string GetTitle(Operation operation)
+        {
+            return EditorFeaturesResources.Toggle_Block_Comment;
+        }
+
+        protected override string GetMessage(Operation operation)
+        {
+            return EditorFeaturesResources.Toggling_block_comment_on_selection;
+        }
+
+        protected override void SetTrackingSpans(ITextView textView, ITextBuffer buffer, List<CommentTrackingSpan> trackingSpans)
+        {
+            var spans = trackingSpans.Select(trackingSpan => trackingSpan.ToSelection(buffer));
+            textView.GetMultiSelectionBroker().SetSelectionRange(spans, spans.Last());
+        }
+
+        internal override void CollectEdits(Document document, ICommentSelectionService service, NormalizedSnapshotSpanCollection selectedSpans,
+            List<TextChange> textChanges, List<CommentTrackingSpan> trackingSpans, Operation operation, CancellationToken cancellationToken)
+        {
+            var experimentationService = document.Project.Solution.Workspace.Services.GetRequiredService<IExperimentationService>();
+            if (!experimentationService.IsExperimentEnabled(WellKnownExperimentNames.RoslynToggleBlockComment))
+            {
+                return;
+            }
+
+            if (selectedSpans.IsEmpty())
+            {
+                return;
+            }
+
+            var getRoot = document.GetSyntaxRootAsync();
+            var commentInfo = service.GetInfoAsync(document, selectedSpans.First().Span.ToTextSpan(), cancellationToken).WaitAndGetResult(cancellationToken);
+            var root = getRoot.WaitAndGetResult(cancellationToken);
+            if (commentInfo.SupportsBlockComment)
+            {
+                ToggleBlockComments(commentInfo, root, selectedSpans, textChanges, trackingSpans);
+            }
+        }
+
+        private static void ToggleBlockComments(CommentSelectionInfo commentInfo, SyntaxNode root, NormalizedSnapshotSpanCollection selectedSpans,
+            List<TextChange> textChanges, List<CommentTrackingSpan> trackingSpans)
+        {
+            var blockCommentedSpans = GetDescendentBlockCommentSpansFromRoot(root);
+            var blockCommentSelectionHelpers = selectedSpans.Select(span => new BlockCommentSelectionHelper(blockCommentedSpans, span));
+
+            // If there is a multi selection, either uncomment all or comment all.
+            var onlyAddComment = false;
+            if (selectedSpans.Count > 1)
+            {
+                onlyAddComment = blockCommentSelectionHelpers.Where(helper => !helper.IsEntirelyCommented()).Any();
+            }
+
+            foreach (var blockCommentSelection in blockCommentSelectionHelpers)
+            {
+                if (!onlyAddComment && TryUncommentBlockComment(blockCommentedSpans, blockCommentSelection, textChanges, trackingSpans, commentInfo))
+                { }
+                else
```

Comment:
?

Reply:
If you're referring to 'onlyAddComment ' In the case of multi caret selection, there can be one selection that is commented, one that is not.  In that case we only comment everything or uncomment everything.

Reply:
i'm referring to the oddness of this code, specifically the empty 'true' part of the if-statement.  This is not a pattern seen elsewhere in the IDE codebase.